packages/wrangler/src/type-generation.ts

```diff
@@ -240,6 +246,43 @@ async function generateTypes(
 		}
 	}
 
+	if (configToDTS.send_email) {
+		for (const sendEmail of configToDTS.send_email) {
+			envTypeStructure.push(`${sendEmail.name}: SendEmail;`);
+		}
+	}
+
+	if (configToDTS.vectorize) {
+		for (const vectorize of configToDTS.vectorize) {
+			envTypeStructure.push(`${vectorize.binding}: VectorizeIndex;`);
+		}
+	}
+
+	if (configToDTS.hyperdrive) {
+		for (const hyperdrive of configToDTS.hyperdrive) {
+			envTypeStructure.push(`${hyperdrive.binding}: Hyperdrive;`);
+		}
+	}
+
+	if (configToDTS.mtls_certificates) {
+		for (const mtlsCertificate of configToDTS.mtls_certificates) {
+			envTypeStructure.push(`${mtlsCertificate.binding}: Fetcher;`);
+		}
+	}
+
+	if (configToDTS.browser) {
+		// The BrowserWorker type in @cloudflare/puppeteer is of type
+		// { fetch: typeof fetch }, but workers-types doesn't include it
+		// and Fetcher is valid for the purposes of handing it to puppeteer
+		envTypeStructure.push(`${configToDTS.browser.binding}: Fetcher;`);
+	}
+
+	if (configToDTS.ai) {
+		// The @cloudflare/ai package types specify any, but from its usage
+		// we can infer it's a Fetcher type (await this.binding.fetch)
+		envTypeStructure.push(`${configToDTS.ai.binding}: Fetcher;`);
```

Comment:
I'd be tempted to keep it as `any` if it is in `@cloudflare/ai` :thinking:

Reply:
I think this should probably be `any`/`unknown` since I'm not sure it's guaranteed to always be a `Fetcher`

Reply:
I'd like to avoid generating `any` into user projects - lots of people use the recommended preset from typescript-eslint which includes [no-explicit-any](https://typescript-eslint.io/rules/no-explicit-any/). `unknown` is somewhat better, I haven't thought of that myself. Let me push a fix

Reply:
yes I would also go with `unknown`, I didn't suggest it as I don't know if `@cloudflare/ai` could complain about that?

Reply:
Fortunately _anything_ can be passed to `any`, including unknown: [ts playground](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABFApgZygCgIYCcDmAXItmAJ4CUiA3gFCIOK0C+ttAJihADZ4qIQEGRADc8MbACNuKYuADWYOAHcwbVBkxjcE6Sgq0gA)

Changed to unknown in [b69e965](https://github.com/cloudflare/workers-sdk/pull/5089/commits/b69e9650bfee4804e4b82e9674afbda5d0d5d0ca)